pkgs/development/libraries/science/networking/ns-3/default.nix

```diff
@@ -1,77 +1,109 @@
 { stdenv
+, breakpointHook
 , fetchFromGitLab
 , python
-, wafHook
-
-# for binding generation
+, libxml2
+, sqlite
+, dpdk
+, openssl
+, gsl
+, libnl
+, sphinx
+, boost
+, gtk3-x11
+, root
+, glib
+, libpcap
+, libelf
+, cmake
+, pkg-config
+, jansson
+, pcre2
+  # for binding generation
 , castxml ? null
 
-# can take a long time, generates > 30000 images/graphs
+  # can take a long time, generates > 30000 images/graphs
 , enableDoxygen ? false
 
-# e.g. "optimized" or "debug". If not set, use default one
-, build_profile ? null
-
-# --enable-examples
-, withExamples ? false
-
-# very long
-, withManual ? false, doxygen ? null, graphviz ? null, imagemagick ? null
-# for manual, tetex is used to get the eps2pdf binary
-# texlive to get latexmk. building manual still fails though
-, dia, tetex ? null, ghostscript ? null, texlive ? null
-
-# generates python bindings
-, pythonSupport ? false, ncurses ? null
-
-# All modules can be enabled by choosing 'all_modules'.
-# we include here the DCE mandatory ones
+  # very long
+, withManual ? false
+, doxygen ? null
+, graphviz ? null
+, imagemagick ? null
+  # for manual, tetex is used to get the eps2pdf binary
+  # texlive to get latexmk. building manual still fails though
+, dia
+, tetex ? null
+, ghostscript ? null
+, texlive ? null
+
+  # generates python bindings
+, pythonSupport ? true
+, ncurses ? null
+
+  # All modules can be enabled by choosing 'all_modules'.
+  # we include here the DCE mandatory ones
 , modules ? [ "core" "network" "internet" "point-to-point" "point-to-point-layout" "fd-net-device" "netanim" ]
 , lib
 }:
 
 let
-  pythonEnv = python.withPackages(ps:
+  pythonEnv = python.withPackages (ps:
     lib.optional withManual ps.sphinx
     ++ lib.optionals pythonSupport (with ps;[ pybindgen pygccxml ])
   );
 in
 stdenv.mkDerivation rec {
   pname = "ns-3";
-  version = "35";
+  version = "39";
 
   src = fetchFromGitLab {
     owner = "nsnam";
-    repo   = "ns-3-dev";
-    rev    = "ns-3.${version}";
-    sha256 = "sha256-3w+lCWWra9sndL8+vkGfH5plrDYYCMFi1PzwIVRku6I=";
+    repo = "ns-3-dev";
+    rev = "ns-3.${version}";
+    sha256 = "sha256-2d8xCCfxRpcCZgt7ne17F7cUo/wIxLyvjQs3izNUnmY=";
   };
 
-  nativeBuildInputs = [ wafHook python ];
+  nativeBuildInputs = [ breakpointHook cmake pkg-config pythonEnv ];
 
-  outputs = [ "out" ] ++ lib.optional pythonSupport "py";
+  outputs = [ "out" ];
```

Comment:
I wont block the merge on this but ideally we would have the manual in a separate output since it's heavy